Plugin fails on Gradle 7.x with "MissingPropertyException: Could not get unknown property 'compile' for configuration container of type org.gradle.api.internal.artifacts.configurations.DefaultConfigurationContainer."

When using version 'com.radcortez.gradle:openjpa-gradle-plugin:3.1.0' of this plugin when I upgrade my project from Gradle 6.9.2 to Gradle 7.6, I get the following error

```
Caused by: groovy.lang.MissingPropertyException: Could not get unknown property 'compile' for configuration container of type org.gradle.api.internal.artifacts.configurations.DefaultConfigurationContainer.
        at org.gradle.internal.metaobject.AbstractDynamicObject.getMissingProperty(AbstractDynamicObject.java:85)
        at org.gradle.internal.metaobject.AbstractDynamicObject.getProperty(AbstractDynamicObject.java:62)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfigurationContainer_Decorated.getProperty(Unknown Source)
        at com.radcortez.gradle.plugin.openjpa.metamodel.MetamodelTask$_closure1.doCall(MetamodelTask.groovy:22)
```


Using a locally compiled plugin-jar with the following patch seems to make it work, but this is more of a hack and not a real solution:

```
Index: src/main/groovy/com/radcortez/gradle/plugin/openjpa/metamodel/MetamodelTask.groovy
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/groovy/com/radcortez/gradle/plugin/openjpa/metamodel/MetamodelTask.groovy b/src/main/groovy/com/radcortez/gradle/plugin/openjpa/metamodel/MetamodelTask.groovy
--- a/src/main/groovy/com/radcortez/gradle/plugin/openjpa/metamodel/MetamodelTask.groovy	(revision 1504f299c17934b67e3adee41f3743216d1ce177)
+++ b/src/main/groovy/com/radcortez/gradle/plugin/openjpa/metamodel/MetamodelTask.groovy	(date 1673732856806)
@@ -19,7 +19,14 @@
             SourceDirectorySet mainJava = project.sourceSets.main.java
             source(mainJava.srcDirs)
 
-            setClasspath(project.configurations.compile)
+            try {
+                setClasspath(project.configurations.compile)
+            } catch (Throwable e) {
+                setClasspath(project.configurations.compileClasspath)
+            }
             setDestinationDir(project.file(configuration.metamodelOutputFolder))
 
             def openjpaConfig = project.configurations.detachedConfiguration(
```

--- a/src/main/groovy/com/radcortez/gradle/plugin/openjpa/metamodel/MetamodelTask.groovy
+++ b/src/main/groovy/com/radcortez/gradle/plugin/openjpa/metamodel/MetamodelTask.groovy
@@ -19,7 +19,7 @@ class MetamodelTask extends JavaCompile {
             SourceDirectorySet mainJava = project.sourceSets.main.java
             source(mainJava.srcDirs)
 
-            setClasspath(project.configurations.compile)
+            setClasspath(project.configurations.compileClasspath)
             setDestinationDir(project.file(configuration.metamodelOutputFolder))
 
             def openjpaConfig = project.configurations.detachedConfiguration(

